juju/model.py

```diff
@@ -0,0 +1,167 @@
+import json
+from collections.abc import Mapping
+from functools import lru_cache
+from subprocess import run, PIPE
+
+
+class Model:
+    def __init__(self, relation_names, backend):
+        self.relations = RelationMap(relation_names, backend)
+        self._backend = backend
+
+    @property
+    def app(self):
+        return self._backend.local_app
+
+    @property
+    def unit(self):
+        return self._backend.local_unit
+
+    def relation(self, relation_name):
+        """Return the first Relation object for the named relation, or None.
+
+        This is a convenience method for the case where only a single related
+        app is supported. If there are more than one remote apps on the relation,
+        it will raise a TooManyRelatedApps model error. If no there are no remote
+        apps on the relation, it will return None. Otherwise, it is equivalent
+        to self.relations[relation_name][0].
+        """
+        num_related = len(self.relations[relation_name])
+        if num_related > 1:
+            raise TooManyRelatedApps(relation_name, num_related, 1)
+        elif num_related == 0:
+            return None
+        else:
+            return self.relations[relation_name][0]
```

Comment:
This might read slightly better if we order it in terms of num_related (0, 1, many).